db/crud.go

```diff
@@ -160,7 +162,15 @@ func (db *DatabaseContext) OnDemandImportForGet(docid string, rawDoc []byte, raw
 	isDelete := rawDoc == nil
 	importDb := Database{DatabaseContext: db, user: nil}
 	var importErr error
-	docOut, importErr = importDb.ImportDocRaw(docid, rawDoc, rawXattr, isDelete, cas, ImportOnDemand)
+
+	// Get the doc expiry
```

Comment:
Now that we support passing the expiry back from inside importDoc, should we just retrieve the expiry there if not specified?  i.e. ImportDoc/ImportDocRaw would take expiry as *uint32.  When nil, it would retrieve from bucket.  When not nil (the DCP case), it wouldn't retrieve from bucket.

I like the idea of keeping all the import-specific logic inside ImportDoc (so that callers don't need to know how to retrieve expiry).

Reply:
I made that change and verified the tests are still passing